### ISSUE TITLE
Bundler version 2.3.7 is not supported

### ISSUE THREAD

**Issue by SerhiDi** [Shopify/cli#594]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

App, Extension

### Expected behavior

Again, errors and mistakes. 
Shopify it's ridiculous.

I can't run a simple shopify app with a theme extension.

### Actual behavior

does not work, but it's shopify

### Stack trace

```shell
━━━━━━ Error ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    Bundler version 2.3.7 is not supported

    What to try:
    To update to the latest version of Bundler, run gem install bundler

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━


AND

━━━━━━ Error ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    Command failed with exit code 1: bundle install
    Your lockfile was created by an old Bundler that left some things out.
    Because of the missing DEPENDENCIES, we can only install gems one at a time, instead of installing 16 at a time.
    You can fix this by adding the missing gems to your Gemfile, running bundle install, and then removing the gems from your Gemfile.
    The missing gems are:
    * mini_portile2 depended upon by nokogiri
    --- ERROR REPORT TEMPLATE -------------------------------------------------------

    
    NoMethodError: undefined method `full_name' for nil:NilClass
```

### Reproduction steps

1. Open shopify docs
https://shopify.dev/apps/getting-started/create
2. Install shopify cli
3. Create an app - `npm init @shopify/app@latest`
4. `npm run dev` - all good. You can see this app
5. Open shopify docs
https://shopify.dev/apps/online-store/theme-app-extensions/getting-started
6. Create an extension - `npm run shopify app generate extension`
7. `npm run dev` - see the error `Bundler version 2.3.7 is not supported`
8. `gem install bundler`
9. `npm run dev` - see the error about missing dependencies - `mini_portile2 depended upon by nokogiri`

### Operating System

Windows 10 x64

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.0.24

### Node version (run `node -v` if you're not sure)

v18.7.0

### What language and version are you using in your application?

Node

**Comment by SerhiDi** [comment]
Ruby+Devkit 3.1.2-1 (x64) - error
Ruby+Devkit 3.0.4-1 (x64) - ok

**Comment by SerhiDi** [comment]
Hello. @karreiro 
I tried these versions and they have different behavior.
Installing steps: ruby + MSYS2 + `gem install bundler`

1. devkit-2.6.10-1 - `npm run dev` - Error `Ruby version 2.6.10 is not supported`. The minimum version is 2.7.5.

2. devkit-2.7.6-1 and devkit-3.0.4-1 - `npm run dev` works and run ngrok. But after a minute it stops and I see:

```
    Command failed with exit code 1: bundle install
    Your lockfile was created by an old Bundler that left some things out.
    Because of the missing DEPENDENCIES, we can only install gems one at a time, instead of installing 16 at a time.
    You can fix this by adding the missing gems to your Gemfile, running bundle install, and then removing the gems from your Gemfile.
    The missing gems are:
    * mini_portile2 depended upon by nokogiri
    --- ERROR REPORT TEMPLATE -------------------------------------------------------

    ```
    NoMethodError: undefined method `full_name' for nil:NilClass
    ```

3. devkit-3.1.2-1 - `npm run dev` immediately displays the error above.

**Comment by thomaswelter** [comment]
I have the exact same problem and I can confirm that the steps provided by @karreiro do not solve it.

**Comment by karreiro** [comment]
@SerhiDi @thomaswelter, thank you for the extra context.

Could you please confirm your Bundler version (with ` bundler --version`)? The version is fine if it's higher than `2.3.18` (otherwise, executing  `gem install bundler -v 2.3.18` solves that).

I've noticed, based on the stack trace above, the old Bundler left an old lockfile. Is there any extra stack trace available? If so, we might locate that file to remove.

(also, I believe that by restarting the machine, this lockfile file should go way -- I'm assuming that based on `XDG_CACHE_HOME` behavior on Windows).

**Comment by SerhiDi** [comment]
```
━━━━━━ Error ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    Command failed with exit code 1: bundle install
    Your lockfile was created by an old Bundler that left some things out.
    Because of the missing DEPENDENCIES, we can only install gems one at a time, instead of installing 16 at a time.
    You can fix this by adding the missing gems to your Gemfile, running bundle install, and then removing the gems from your Gemfile.
    The missing gems are:
    * mini_portile2 depended upon by nokogiri
    --- ERROR REPORT TEMPLATE -------------------------------------------------------

    ```
    NoMethodError: undefined method `full_name' for nil:NilClass
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:124:in `block (2 levels) in check_for_unmet_dependencies'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:123:in `each'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:123:in `block in check_for_unmet_dependencies'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:122:in `each'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:122:in `check_for_unmet_dependencies'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:100:in `call'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:71:in `call'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer.rb:254:in `install_in_parallel'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer.rb:209:in `install'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer.rb:89:in `block in run'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/process_lock.rb:12:in `block in lock'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/process_lock.rb:9:in `open'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/process_lock.rb:9:in `lock'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer.rb:71:in `run'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer.rb:23:in `install'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/cli/install.rb:62:in `run'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/cli.rb:257:in `block in install'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/settings.rb:131:in `temporary'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/cli.rb:256:in `install'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/cli.rb:31:in `dispatch'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/cli.rb:25:in `start'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/exe/bundle:48:in `block in <top (required)>'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'
      C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/exe/bundle:36:in `<top (required)>'
      C:/Ruby27-x64/bin/bundle:23:in `load'
      C:/Ruby27-x64/bin/bundle:23:in `<main>'

    ```

    ## Environment

    ```
    Bundler       2.3.23
      Platforms   ruby, x64-mingw32
    Ruby          2.7.6p219 (2022-04-12 revision c9c2245c0a25176072e02db9254f0e0c84c805cd) [x64-mingw32]
      Full Path   C:/Ruby27-x64/bin/ruby.exe
      Config Dir  C:/ProgramData
    RubyGems      3.1.6
      Gem Home    C:/Users/{{USER_NAME}}/AppData/Local/shopify-cli-nodejs/Cache/vendor/ruby-cli/2.25.0/ruby/2.7.0
      Gem Path    C:/Users/{{USER_NAME}}/AppData/Local/shopify-cli-nodejs/Cache/vendor/ruby-cli/2.25.0/ruby/2.7.0
      User Home   C:/Users/{{USER_NAME}}
      User Path   C:/Users/{{USER_NAME}}/.gem/ruby/2.7.0
      Bin Dir     C:/Users/{{USER_NAME}}/AppData/Local/shopify-cli-nodejs/Cache/vendor/ruby-cli/2.25.0/ruby/2.7.0/bin
    OpenSSL
      Compiled    OpenSSL 1.1.1n  15 Mar 2022
      Loaded      OpenSSL 1.1.1n  15 Mar 2022
      Cert File   C:/Ruby27-x64/ssl/cert.pem
      Cert Dir    C:/Ruby27-x64/ssl/certs
    Tools
      Git         2.35.1.windows.2
      RVM         not installed
      rbenv       not installed
      chruby      not installed
    ```

    ## Bundler Build Metadata

    ```
    Built At          2022-10-05
    Git SHA           250d9d485d
    Released Version  true
    ```

    ## Bundler settings

    ```
    path
      Set for your local app (C:/Users/{{USER_NAME}}/AppData/Local/shopify-cli-nodejs/Cache/vendor/ruby-cli/2.25.0/.bundle/config): "C:/Users/{{USER_NAME}}/AppData/Local/shopify-cli-nodejs/Cache/vendor/ruby-cli/2.25.0"
    ```

    ## Gemfile

    ### Gemfile

    ```ruby
    source 'https://rubygems.org'
    gem 'shopify-cli', '2.25.0'
    ```

    ### Gemfile.lock

    ```
    <No C:/Users/{{USER_NAME}}/AppData/Local/shopify-cli-nodejs/Cache/vendor/ruby-cli/2.25.0/Gemfile.lock found>
    ```

    --- TEMPLATE END ----------------------------------------------------------------

    Unfortunately, an unexpected error occurred, and Bundler cannot continue.

    First, try this link to see if there are any existing issue reports for this error:
    https://github.com/rubygems/rubygems/search?q=undefined+method+%60full_name%27+for+nil+NilClass&type=Issues

    If there aren't any reports for this error yet, please fill in the new issue form located at https://github.com/rubygems/rubygems/issues/new?labels=Bundler&template=bundler-related-issue.md, and copy and paste the report template above in there.
    Fetching gem metadata from https://rubygems.org/.............
    Resolving dependencies...
    Fetching ast 2.4.2
    Installing ast 2.4.2
    Fetching concurrent-ruby 1.1.10
    Installing concurrent-ruby 1.1.10
    Fetching bugsnag 6.24.2
    Installing bugsnag 6.24.2
    Using bundler 2.3.23
    Fetching ffi 1.15.5
    Installing ffi 1.15.5 with native extensions
    Fetching liquid 5.4.0
    Installing liquid 5.4.0
    Fetching rb-fsevent 0.11.2
    Installing rb-fsevent 0.11.2
    Fetching rb-inotify 0.10.1
    Installing rb-inotify 0.10.1
    Fetching listen 3.7.1
    Installing listen 3.7.1
    Fetching racc 1.6.0
    Installing racc 1.6.0 with native extensions
    Fetching nokogiri 1.13.8
    Installing nokogiri 1.13.8 with native extensions

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

**Comment by karreiro** [comment]
Thank you for the extra details, @SerhiDi. With them, I could reproduced the same issue:

![image](https://user-images.githubusercontent.com/1079279/196128766-19b2cf21-6f66-4c62-ab1f-414260be8d07.png)


Here are the steps I've followed to workaround that:

- Uninstall all Ruby versions in the Control Panel
- Restart the machine
- [Install](https://rubyinstaller.org/downloads/) Ruby 3.0.4
- Select the MSYS2 component and select the MYS2 base installation
- Execute `gem install bundler`

Please let me know if they work for you. If so, we will investigate how to improve this use case out of the box and cover this scenario in the documentation.

**Comment by SerhiDi** [comment]
Hello @karreiro,
I wrote earlier that I checked all this on version 3.0.4. There was that error but after a while.

But now I can't even start the server:
```
Error: connect ECONNREFUSED ::1:50438
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1247:16) {
  errno: -4078,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 50438
}
```


![image](https://user-images.githubusercontent.com/10574421/196175830-3bcc728e-ca47-4992-97af-d6c7178b6a3a.png)

Similar issues:
[1](https://github.com/Shopify/shopify-app-template-node/issues/1117) [2](https://github.com/Shopify/cli/issues/630)


Constantly with shopify some problems,
And the creators of the shopify tools do not want to deal with problems.
I do not know any other system or service where such nonsense can be found
